overlord/snapshotstate/backend/backend.go

```diff
@@ -157,6 +157,42 @@ func Filename(snapshot *client.Snapshot) string {
 	return filepath.Join(dirs.SnapshotsDir, fmt.Sprintf("%d_%s_%s_%s.zip", snapshot.SetID, snapshot.Snap, snapshot.Version, snapshot.Revision))
 }
 
+// EstimateSnapshotSize calculates estimated size of the snapshot.
+// XXX: this doesn't take users into account yet, because it is used only
+// for automatic snapshots at the moment.
+func EstimateSnapshotSize(si *snap.Info) (uint64, error) {
+	var total uint64
+	calculateSize := func(path string, finfo os.FileInfo, err error) error {
+		if finfo.Mode().IsRegular() {
+			total += uint64(finfo.Size())
+		}
+		return err
+	}
+
+	visitDir := func(dir string) error {
+		exists, isDir, err := osutil.DirExists(dir)
+		if err != nil {
+			return err
+		}
+		if !(exists && isDir) {
+			return nil
+		}
+		return filepath.Walk(dir, calculateSize)
+	}
+
+	for _, dir := range []string{si.DataDir(), si.CommonDataDir()} {
+		if err := visitDir(dir); err != nil {
+			return 0, err
+		}
+	}
+
+	// XXX: if we want to support users, then we need to iterate over them,
+	// similarly to Save() below.
```

Comment:
we snapshot users when we remove snaps, no? so we need to do this I think

Reply:
Hmm, you're right, I misread users handling code, it defaults to all users if not specified, so yes automatic snapshots include them. Thanks!

Reply:
Done.